scipy/optimize/_remove_redundancy.py

```diff
@@ -192,28 +193,27 @@ def _remove_redundancy_dense(A, rhs):
             j = b[i-1]
             lu = bg_update_dense(lu, perm_r, A[:, j], i-1)
         except Exception:
```

Comment:
This isn't blocking, but, could we try to avoid generic exceptions?

Reply:
Yes, I strive for that in general. If I remember correctly: here, as in `_get_solver` of [`_linprog_ip.py`](https://github.com/scipy/scipy/blob/master/scipy/optimize/_linprog_ip.py), there are a variety of exceptions that can be raised when factoring, and I wasn't sure whether I had them all. I should revisit this, so I've added it to gh-9671.